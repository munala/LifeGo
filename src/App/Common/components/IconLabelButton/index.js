import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

import './styles.css';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
  },
  iconSmall: {
    fontSize: 20,
  },
});

const IconLabelButton = ({
  label,
  name,
  classes,
  style,
  onClick,
  disabled,
}) => (
  <div className="container" >
    <Button
      className={classes.button}
      color="primary"
      variant="contained"
      onClick={onClick}
      disabled={disabled}
      style={style}
    >
      {label}
      <Icon
        color="action"
        className={classes.rightIcon}
      >{name}
      </Icon>
    </Button>
  </div>
);

IconLabelButton.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.shape({}),
  classes: PropTypes.shape({}).isRequired,
};

IconLabelButton.defaultProps = {
  disabled: false,
  style: undefined,
};

export default withStyles(styles)(IconLabelButton);
