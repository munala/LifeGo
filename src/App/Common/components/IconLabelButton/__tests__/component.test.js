import React from 'react';
import { shallow } from 'enzyme';
import IconLabelButton from '../';

describe('IconLabelButton tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<IconLabelButton
      label="source"
      name="home"
      onClick={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
