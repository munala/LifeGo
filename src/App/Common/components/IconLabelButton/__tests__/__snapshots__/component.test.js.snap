// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLabelButton tests renders without crashing 1`] = `
<IconLabelButton
  classes={
    Object {
      "button": "IconLabelButton-button-1",
      "iconSmall": "IconLabelButton-iconSmall-3",
      "rightIcon": "IconLabelButton-rightIcon-2",
    }
  }
  disabled={false}
  label="source"
  name="home"
  onClick={[Function]}
/>
`;
